cypress/integration/User.spec.ts

```diff
@@ -23,13 +27,13 @@ describe('Unauthenticated user', () => {
 
 describe('Authenticated user', () => {
   beforeEach(() => {
-    cy.visit(
-      `/target/${Cypress.env('target-id')}?${querystring.stringify(authObject)}`
-    );
+    cy.visitTargetPage(Cypress.env('target-id'));
   });
 
   it('Has username in header', () => {
     cy.get('header').first().should('not.contain', 'Kirjaudu');
-    cy.get('header').first().should('contain', authObject.username);
+    cy.get('header')
+      .first()
+      .should('contain', parseUserName(Cypress.env('AUTH_QUERY_PARAMETERS')));
```

Comment:
Do we need to parse the user name in this? Could we just hard-code the value here? It feels like it adds a possibility for an error. If our test function `parseUserName` works wrong, our test might fail, even our application code would work ok. Just not sure if I like that our tests depends heavily on our test function itself. Could we somehow apply the exact same function that our react-router is using? :D Probably not. :thinking:

Reply:
Good points. I'll think of something else 👍

Reply:
Could one solution be that we use data-* attributes that cypress recommends? https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements

In short, instead of checking match for the actual name, we would check the existence of an element that should only appear after successful authentication. Or we could check that it needs to have some content in it. The use of data-* attributes feels most clean solution though

Reply:
Hmm.. I can check if it is possible